rtrlib/rtr/packets.c

```diff
@@ -1049,6 +1094,55 @@ static int rtr_update_spki_table(struct rtr_socket *rtr_socket, struct spki_tabl
 	return RTR_SUCCESS;
 }
 
+static int rtr_update_aspa_table(struct rtr_socket *rtr_socket, struct aspa_table *aspa_table, const void *pdu)
+{
+	const enum pdu_type type = rtr_get_pdu_type(pdu);
+
+	assert(type == ASPA);
+
+	struct aspas_record entry;
+
+	rtr_aspa_pdu_2_aspa_record(rtr_socket, pdu, &entry, type)
+
+	int rtval;
+	// TODO: are there any other flags different from 0x01?
+	if (((struct pdu_aspa *)pdu)->flags & 1 == 1) {
+		rtval = aspa_table_add_entry(aspa_table, &entry);
```

Comment:
do i assume correctly that you implemented against a not-yet-existing interface here?